frontend/src/features/SidebarSuite/uiSettings/DbViewSelector.tsx

```diff
@@ -22,14 +46,21 @@ export const DbViewSelector = () => {
 
   const availableViews = useAvailableDbViews();
 
-  const handleChange = async (e: SelectChangeEvent) => {
-    const newView = getValidDbView(e.target.value);
-    await router.push({
-      pathname: router.pathname.replace(currentView, newView),
-      query: { ...router.query },
-    });
-    setCurrentDbView(newView);
-  };
+  const handleChange = React.useCallback(
+    async (e: SelectChangeEvent) => {
+      const newView = getValidDbView(e.target.value);
+      const queryParams = getViewQueryParams(newView);
+
+      const { language, file } = router.query;
+
+      await router.push({
+        pathname: router.pathname.replace(currentView, newView),
```

Comment:
I would use `push` instead of replace here to preserve the go back browser functionality

Reply:
:confused: .... mmmm.... this is using `push` (the replace is a string replace to update the path)

Reply:
ah, sorry, I think I had tunnel vision when writing this.